genie-common/src/main/java/com/netflix/genie/common/dto/ClusterCriteria.java

```diff
@@ -51,19 +51,16 @@
      */
     @JsonCreator
     public ClusterCriteria(
-        @JsonProperty("tags")
-        final Set<String> tags
+        @JsonProperty("tags") final Set<String> tags
     ) {
         final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
-        if (tags != null) {
-            tags.forEach(
-                tag -> {
-                    if (StringUtils.isNotBlank(tag)) {
-                        builder.add(tag);
-                    }
+        tags.forEach(
+            tag -> {
+                if (StringUtils.isNotBlank(tag)) {
+                    builder.add(tag);
                 }
-            );
-        }
+            }
+        );
         this.tags = builder.build();
```

Comment:
Alternatively:

```java
this.tags = tags.stream().filter(StringUtils::isNotBlank).collect(ImmutableSet.toImmutableSet());
```

Reply:
Would love to but this is only in Guava 21+ and I have concerns about upgrading such a core library out of band with what all the other libraries work with it

Reply:
`tags.forEach` will throw NPE if tags is null. Am I missing something that guarantees this is not null?

Reply:
`@ParametersAreNonnullByDefault` was added to the package header for `com.netflix.genie.common.dto`. As tags are not optional in this case it will ensure something is passed in at compile time.